src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -315,9 +283,11 @@ protected override void Free()
             BoundNode node,
             Binder binder,
             Conversions conversions,
-            ArrayBuilder<(BoundReturnStatement, TypeWithAnnotations)> returnTypesOpt,
             VariableState initialState,
-            Dictionary<BoundExpression, (NullabilityInfo, TypeSymbol)> analyzedNullabilityMapOpt)
+            ImmutableArray<(BoundReturnStatement, TypeWithAnnotations)> returnTypesOpt,
+            Dictionary<BoundExpression, (NullabilityInfo, TypeSymbol)> analyzedNullabilityMapOpt,
+            bool takeIncrementalSnapshots,
+            Snapshotter snapshotter = null)
```

Comment:
I think it would be better to not add both parameters to this method, otherwise how would we make sure they agree with each other.

Reply:
Initially I had no choice, as the Snapshotter was used both when taking incremental snapshots and for getting the variable state to pass to the child method. That has been factored out and we just pass the manager.

---
In reply to: [288780361](https://github.com/dotnet/roslyn/pull/35850#discussion_r288780361) [](ancestors = 288780361)